### PR TITLE
Binary patcher generalization

### PR DESCRIPTION
**Pull request by TAImatem**
generalized BinaryPatcher and moved VR code to separate class

### CODE REVIEW

File: OWML.Patcher/BinaryPatcher.cs [hunk 1]
```diff
@@ -11,152 +10,149 @@ public class BinaryPatcher
...
-                _writer.WriteLine("globalgamemanagers already patched.");
-                return;
+                return new byte[0];
             }
 
-            BackupFile(filePath);
-            var patchedBytes = CreatePatchedFileBytes(fileBytes, patchStartIndex);
-            File.WriteAllBytes(filePath, patchedBytes);
-            _writer.WriteLine("Successfully patched globalgamemanagers.", MessageType.Success);
+            (var sectorStart, var sectorSize) = GetSectorInfo(fileBytes, sector);
```
Comment (amazingalek) [inline OWML.Patcher/BinaryPatcher.cs:70]: `(var sectorStart, var sectorSize)` -> `var (sectorStart, sectorSize)` (a couple places)

**Comment by amazingalek** [comment]
var instead of explicit types multiple places

File: OWML.Patcher/VrPatcher.cs [hunk 2]
```diff
@@ -8,6 +8,7 @@ public class VRPatcher
         private readonly IOwmlConfig _owmlConfig;
         private readonly IModConsole _writer;
```
Comment (amazingalek) [inline OWML.Patcher/VrPatcher.cs:9]: No longer needed

Comment (TAImatem) [reply OWML.Patcher/VrPatcher.cs:9]: _owmlConfig seems to be used for .dll copying

File: OWML.Patcher/BinaryPatcher.cs [hunk 3]
```diff
@@ -11,152 +10,149 @@ public class BinaryPatcher
...
+        public byte[] PatchSectorBytes(byte[] fileBytes, byte[] newBytes, int sector)
         {
-            var existingPatchBytes = Encoding.ASCII.GetBytes(EnabledVRDevice);
-            var existingPatchMatch = 0;
+            if (sector < 0 || sector >= SectorCount)
+            {
+                return fileBytes;
+            }
+
+            (var sectorStart, var sectorSize) = GetSectorInfo(fileBytes, sector);
```
Comment (amazingalek) [inline OWML.Patcher/BinaryPatcher.cs:108]: (var sectorStart, var sectorSize) -> var (sectorStart, sectorSize)

**Comment by amazingalek** [comment]
Did you test that VR patching still works?

File: OWML.Patcher/VRFilePatcher.cs [hunk 4]
```diff
@@ -0,0 +1,106 @@
...
+                else
+                {
+                    patchZoneMatch = 0;
+                }
+                if (patchZoneMatch == patchZoneBytes.Length)
+                {
+                    return i + PatchStartZoneOffset;
+                }
+            }
+            throw new Exception("Could not find patch zone in globalgamemanagers. This probably means the VR patch needs to be updated.");
```
Comment (amazingalek) [inline OWML.Patcher/VRFilePatcher.cs:67]: What happens in this case? Does this become a Unity error?

Comment (TAImatem) [reply OWML.Patcher/VRFilePatcher.cs:67]: not really my code, not really sure 🤷‍♂️

**Pull request merged**